.gitignore

```diff
@@ -1,68 +1,60 @@
 # Don't add any custom patterns for platform-specific files like .DS_Store
 # or generated by custom development tools like .vscode or .idea that don't
 # affect other developers. Instead add those custom patterns to your global
-# .gitignore file.
+# gitignore or to your local .git/info/exclude.
 
+# Logs, local, temp files
 *.diff
 *.log
+mixxx.log*
 *.o
 *.orig
 *.os
 *.patch
-!build/appveyor/NSIS.patch
 *.pyc
 *.rej
 *_build
 *~
 .#*
 
-dist32
-dist64
-
+# Compilation artifacts
 src/mixxx.rc.include
 src/mixxx.res
+src/test/**/*.actual
 
-.sconf_temp
-.sconsign.branch
-.sconsign.dblite
-
-*.sqlite
-
-*.cfg
-!/res/keyboard/*.cfg
-
-cache
-
-lib/*/*.a
-lib/*/lib/*.a
+res/qrc_mixxx.cc
 
 mixxx
 mixxx-test
 
-src/test/test_data
-src/test/golden_buffers/*.actual
-
-res/qrc_mixxx.cc
+# ???
+*.obj
+*.pdb
+lib/**/*.lib
+lib/**/*.a
 
+# ???
 *.wixobj
 *.wixpdb
 *.msi
 *.mst
 *.cab
 build/wix/subdirs/*.wxs
```

Comment:
All of this is obsolete IMHO. That stuff should be part of the `cmake_build` directory, not the repo root anymore.